aiu_fms_testing_utils/utils/direct_quantization.py

```diff
@@ -0,0 +1,260 @@
+                qcfg["smoothq_alpha"] = 0
+                print("[WARNNG] using smoothq_alpha = 0 for scale generation")
+            act_scales = get_act_scales(model, dq_dataloader, qcfg, device="cpu")
+            torch.save(act_scales, qcfg['act_scale_path'])
+            if smoothq_alpha_requested:
+                qcfg["smoothq_alpha"] = smoothq_alpha_requested
+                print(f"smoothq_alpha set back to {qcfg['smoothq_alpha']}")
+        else:
+            print(
+                f"using smoothq activation scales from {qcfg['act_scale_path']}"
+            )
+            act_scales = torch.load(qcfg['act_scale_path'], map_location='cpu')
+
+        dq_llm(model, act_scales, qcfg)
+        print(f"* SmoothQuant scales applied (took = {time.time() - sq_time_start:.1f} s).")
+        print("=="*20)
+    else:
+        print("* SmoothQuant is DISABLED.")
+
+    #-------------
+    # Run calibration = Direct Quantization DQ
+    #-------------
+    if qcfg['qmodel_calibration_new'] > 0:
+        calib_time_start = time.time()
+        print("* Begin calibration of activation quantized parameters.")
+        pbar = tqdm(
+            dq_dataloader,
+            desc="* Calibration progress",
+            total = qcfg['qmodel_calibration_new']
+            )
+        for data_mb, _ in zip(pbar, range(qcfg['qmodel_calibration_new'])):
+            data_mb = prepare_input(
+                device=model.device,
+                data=data_mb,
+            )
+            with torch.no_grad():
+                model(**data_mb)
+        print(f"* Calibration completed (took = {time.time() - calib_time_start:.1f} s).")
+
+    if args.verbose:
+        print("=" * 60)
+        print("* PARAMETERS")
+        print("\n".join(
+            f"{k:80} {str(list(v.size())):15} {v.dtype}"
+            for k,v in model.named_parameters()
+        ))
+        print("* BUFFERS")
+        print("\n".join(
+            f"{k:80} {str(list(v.size())):15} {v.dtype}"
+            for k,v in model.named_buffers()
+        ))
+
+    #-------------
+    # Save checkpoint with integer weights (AIU requirement)
+    #-------------
+    keys_to_ignore = [
+        "num_module_called",
+        "smoothq_act_scale",
+        "smoothq_alpha",
+        "calib_counter",
+        "obsrv_clipval",
+        "obsrv_clipvaln",
+        "obsrv_w_clipval",
+    ]
+
+    print(f"Begin processing model state dictionary for saving.")
+    new_sd = process_state_dict(
+        model=model,
+        quantized_layers=QUANTIZED_LAYERS_ROBERTA,
+        keys_to_ignore=keys_to_ignore,
+        verbose=args.verbose,
+    )
+
+    task = "mlm" if args.architecture == "roberta" else "qa"
+    smoothq_str = qcfg['smoothq_alpha'] if qcfg['smoothq'] else "no"
+    save_path = str(
+        Path(args.output_path) /
+        f"roberta-base_{task}_w8-{qcfg['qw_mode']}_a8-{qcfg['qa_mode']}"
+        f"_bmm32_smoothq-{smoothq_str}_dq.pt"
+    )
+    torch.save(new_sd, save_path)
```

Comment:
do we want to use safetensors to save the models instead? especially given all the CVEs around torch.load/save lately